### PR TITLE
Set plugin `mode=disabled`(instead of removing whole plugin) when running `qlty plugins disable PLUGIN`

### PR DESCRIPTION
**Pull request by noahd1**
- Changes behavior of the `qlty plugins disable PLUGIN` to set the plugin mode to `disabled` (instead of removing it from the qlty.toml)
- Adds a couple error checks: errors if no plugins in qlty.toml or if the plugin referenced in the command is not found in their qlty.toml (used `upgrade` command as reference)

Couple notes:

1. The conversion of `IssueMode` to a string duplicates the serialization macro strings (`#[serde(rename = "block")]` e.g.) defined in the struct itself. From what I could tell, if we want to avoid this duplication, we could either a) use `serde_json` to translate the IssueMode value to JSON, and then extract the string out of the JSON (yuck). Or, instead, remove these macros and define our own serialization, using the new `to_string` method defined. I'm less sure where the serialization is actually happening, so not confident I could do this.
2. I looked/tested how `qlty plugins enable PLUGIN` behaves. If a plugin is disabled (with the `disable` command or by hand), you get an error message "plugin PLUGIN already enabled". So we'll have to change this behavior to just remove the disabled mode line (I'm guessing).

### CODE REVIEW

File: qlty-cli/src/commands/plugins/disable.rs
Comment (brynary) [inline qlty-cli/src/commands/plugins/disable.rs:33]: Do you see any concrete downside to removing this error check and letting it fall to the second error check? This one seems redundant at first read.

File: qlty-config/src/config/plugin.rs
Comment (brynary) [inline qlty-config/src/config/plugin.rs:718]: This formatting is wrong. Rust has a canonical auto formatted which you should be able to get your editor to run on save.

File: qlty-cli/src/commands/plugins/disable.rs
Comment (brynary) [inline qlty-cli/src/commands/plugins/disable.rs:145]: I assume this is repeating existing pattern but it would be nice if we can test this without touching the filesystem since it's basically a pure function (toml string in, toml string out).

If you were so inclined, it would be a nice refactor

**Review by brynary — APPROVED** [review]
Approving with some notes.

Good catch on "enable" subcommand! It will need to get smarter. 

There's a few cases that it will need to handle... I do think we should retain the error from the "enable" if the plugin is already enabled but it will need to be more complex than just looking for the key

File: qlty-cli/src/commands/plugins/disable.rs
Comment (noahd1) [reply qlty-cli/src/commands/plugins/disable.rs:33]: No. I don't actually. I copied this from the `upgrade` sub-command, which follows similar logic, but will remove here (and note we can/should probably remove from there as well)

File: qlty-cli/src/commands/plugins/disable.rs
Comment (lsegal) [inline qlty-cli/src/commands/plugins/disable.rs:37]: It might be a better pattern to `return Ok(())` from here and `bail()` at the end of the function as a catch-all, allowing you to drop the mutable book-keeping variable.

File: qlty-cli/src/commands/plugins/disable.rs
Comment (noahd1) [reply qlty-cli/src/commands/plugins/disable.rs:141]: I don't want to hold up this PR on this particular refactor. I may look into it, but had started work on updating our `enable` subcommand and want to address that first.

**Pull request merged**
